js/components/contactList/contactList_controller.js

```diff
@@ -23,24 +41,41 @@ app.controller('contactlistCtrl', ['$scope', '$filter', '$route', '$routeParams'
 	$scope.$watch('ctrl.routeParams.uid', function(newValue) {
 		if(newValue === undefined) {
 			// we might have to wait until ng-repeat filled the contactList
-			if(ctrl.contactList.length > 0) {
+			if(ctrl.contactList && ctrl.contactList.length > 0) {
 				$route.updateParams({
 					gid: $routeParams.gid,
 					uid: ctrl.contactList[0].uid()
 				});
 			} else {
 				// watch for next contactList update
-				var unbindWatch = $scope.$watch('ctrl.contactList', function(newValue) {
-					$route.updateParams({
-						gid: $routeParams.gid,
-						uid: ctrl.contactList[0].uid()
-					});
+				var unbindWatch = $scope.$watch('ctrl.contactList', function() {
+					if(ctrl.contactList && ctrl.contactList.length > 0) {
+						$route.updateParams({
+							gid: $routeParams.gid,
+							uid: ctrl.contactList[0].uid()
+						});
+					}
 					unbindWatch(); // unbind as we only want one update
 				});
 			}
 		}
 	});
 
+	$scope.$watch('ctrl.routeParams.gid', function() {
```

Comment:
We now have two `$watch`es on `ctrl.routeParams`. This could probably be reduced to only one.
But let's refactor later.